### PR TITLE
added task add_member_page

### PR DESCRIPTION
**Pull request by patmbolger**
PT Story: https://www.pivotaltracker.com/story/show/145192549

Changes proposed in this pull request:
1. Add task to create a new (empty) file for a new member (static) page

Screenshots (Optional):
---
### Current Member Page list:
screen shot 2017-06-05 at 11 23 26 am

---

### Create new member page file:
```
rake shf:add_member_page[new_member_page]
 ....
[SHF_TASK] [shf:add_member_page] [info] Started at 2017-06-05 11:24:17 -0400
[SHF_TASK] [shf:add_member_page] [info] Created member page file: new_member_page.html
[SHF_TASK] [shf:add_member_page] [info] Finished at 2017-06-05 11:24:17 -0400.
[SHF_TASK] [shf:add_member_page] [info] Duration: 0.0 seconds.
[SHF_TASK] [shf:add_member_page] [info] Information was logged to: log/shf_tasks
```

---

### New Member Page list:
screen shot 2017-06-05 at 11 26 25 am

---

### Edit title and contents of new page:
screen shot 2017-06-05 at 11 27 01 am

Ready for review:
@weedySeaDragon @RobertCram @thesuss

### CODE REVIEW

**Review by RobertCram — COMMENTED** [review]
It is possible to use this rake task to put files anywhere in the directory structure. Should we perhaps add a sanitising check on the use of '/' in the filename?

File: lib/tasks/shf.rake [hunk 1]
```diff
@@ -198,6 +198,31 @@ namespace :shf do
...
+
+    ActivityLogger.open(LOG_FILE, 'SHF_TASK', task_name) do |log|
+
+      filename = args.fetch(:filename) do |_key|
+        log.record('error', 'You must specify a file name')
+        raise 'ERROR: You must specify a file name'
+      end
+
+      # Add html file type if not present
+      filename = filename + '.html' unless filename =~ /.*\.html$/
```
Comment (weedySeaDragon) [inline lib/tasks/shf.rake]: agree with @RobertCram -- check for "risky" characters in the filename and just throw an error as discussed in our scrum.

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]

**Comment by patmbolger** [comment]
I added a check on filename characters - the only acceptable characters are limited to [a-zA-Z0-9_-.].

This is a reasonable set of characters for the member page filename.  The actual title of the page, of course, can be edited to something useful (title defaults to filename otherwise).

File: lib/tasks/shf.rake (hunk 1 above)
Comment (weedySeaDragon) [reply lib/tasks/shf.rake]: How about a slightly more informative error message:  `ERROR:  Unacceptable character(s) in the filename.  The acceptable characters are A-Z (upper- or lowercase), 0-9, '-' and '.'`

Comment (weedySeaDragon) [reply lib/tasks/shf.rake]: If you can make the error message more informative, then this looks good to me.

_(I should have made my previous comment above part of the review.)_

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]

**Comment by patmbolger** [comment]
I made the logged messages a little clearer for the case of invalid characters in filename.

(note that the `record` method (log instance method) writes to both a log file as well as STDOUT).

The user would see something like this:

```
rake shf:add_member_page['file/;;;/name']

....

[SHF_TASK] [shf:add_member_page] [info] Started at 2017-06-09 17:26:45 -0400
[SHF_TASK] [shf:add_member_page] [error] Unacceptable characters in filename: file/;;;/name
[SHF_TASK] [shf:add_member_page] [error] Acceptable characters are a-z, A-Z, 0-9, '_', '-' and '.'
[SHF_TASK] [shf:add_member_page] [info] Finished at 2017-06-09 17:26:45 -0400.
[SHF_TASK] [shf:add_member_page] [info] Duration: 0.0 seconds.
[SHF_TASK] [shf:add_member_page] [info] Information was logged to: log/shf_tasks
rake aborted!
ERROR: Unacceptable filename
```

**Review by weedySeaDragon — APPROVED** [review]

**Review by RobertCram — APPROVED** [review]

**Pull request merged**